### DISCUSSION TITLE
Bagaimana cara menampilkan top n pada ggplot dengan data yang grouping 2 kolom dan summarise count ?

### DISCUSSION THREAD

**Discussion by vidric** [teamalgoritma/community#554] [Q&A] [answered]
Sy mempunyai data dengan grouping sebagai berikut:

plot1 <- battle_card_clean %>%
  filter(win_lose %in% "win", summoner_monster %in% "Summoner") %>%
  group_by(summoner_monster_name, leaderboard) %>%
  summarise(count = n()) %>% 
  ungroup() %>%
  arrange(-count)

Kemudian, ditampilkan pada ggplot seperti ini:
plot_top_summoners <- ggplot(data = plot1, mapping = aes(x = count, y = reorder(summoner_monster_name, count))) +
        geom_col(mapping = aes(fill = leaderboard), position = "stack") +
        labs(title = "Summoners Ranks",
             subtitle = "Based on Win",
             fill = "League",
             x = "Win",
             y = "Summoners") +
        theme_minimal() +
        theme(legend.position = "none")

Hasil:
![Screen Shot 2023-02-21 at 10 55 14](https://user-images.githubusercontent.com/54578949/220243689-8b46cc8c-7ae2-44ef-990e-3ce25799eff2.png)

Yg ingin sy tanyakan, bagaimana cara menampilkan top 10 saja pada ggplot di atas? Terima kasih sebelumnya

**Comment by divakartika** [comment, marked as answer]
Dear @vidric ,

Untuk mendapatkan top 10 dapat dilakukan top_n(10), sehingga code wrangling menjadi seperti berikut

```
plot1 <- battle_card_clean %>%
filter(win_lose %in% "win", summoner_monster %in% "Summoner") %>%
group_by(summoner_monster_name, leaderboard) %>%
summarise(count = n()) %>%
ungroup() %>%
arrange(-count) %>%
top_n(10)
```

Dapat dicoba terlebih dahulu, semoga jawaban ini membantu :)

Best Regards,
Team Algoritma

**Reply by vidric** [reply]
hasilnya seperti ini:
![Screen Shot 2023-02-21 at 11 49 43](https://user-images.githubusercontent.com/54578949/220250512-2f0a175e-14d8-453d-af76-417336106644.png)

**Reply by divakartika** [reply]
Pertama-tama kita dapat membuat dataframe yang difilter untuk memudahkan proses selanjutnya

```
win_summoner <- battle_card_clean %>%
filter(win_lose %in% "win", summoner_monster %in% "Summoner")
```

Karena kita ingin mengambil top 10 `summoner_monster_name` dengan jumlah frekuensi terbanyak, tanpa memandang leaderboard nya, maka pertama perlu dilakukan penghitungan frekuensi `summoner_monster_name` dengan `groupby()` atau `table()`

```
top10 <-  win_summoner %>%
group_by(summoner_monster_name) %>%
summarise(count = n()) %>%
ungroup() %>%
top_n(10)
```

Selanjutnya kita filter data `win_summoner` yang hanya berisi top 10 `summoner_monster_name`

```
plot1 <- win_summoner %>% 
  filter(summoner_monster_name %in% top10$summoner_monster_name) %>% 
  group_by(summoner_monster_name, leaderboard) %>% 
  summarise(count = n()) %>% 
  ungroup()
```

Setelah itu dapat dilakukan plotting

```{r}
ggplot(data = plot1, mapping = aes(x = count, y = reorder(summoner_monster_name, count))) +
geom_col(mapping = aes(fill = leaderboard), position = "stack")
```

Dapat dicoba terlebih dahulu, semoga jawaban ini membantu :)

Best Regards,
Team Algoritma

**Reply by vidric** [reply]
Sy juga sudah membuat ini di awal. Sy kira ada fungsi di ggplot untuk ambil top n.

Terima kasih Mba Diva

**Reply by divakartika** [reply]
Betul Pak David, bergantung pada bagian data wrangling nya. Alternatif code lain yang lebih singkat:

```
plot1 <- win_summoner %>% 
  group_by(summoner_monster_name, leaderboard) %>% 
  summarise(count = n()) %>% 
  ungroup() %>% 
  mutate(summoner_monster_name = reorder(summoner_monster_name, count, decreasing = TRUE)) %>% 
  filter(summoner_monster_name %in% levels(summoner_monster_name)[1:10])
```

credit: @jafarhf1 

Best Regards,
Team Algoritma
